pkg/porter/credentials.go

```diff
@@ -106,22 +107,91 @@ func (p *Porter) GenerateCredentials(opts CredentialOptions) error {
-	bundle, err := p.CNAB.LoadBundle(opts.CNABFile)
 
+	_, err = p.generateNewCredentialSet(opts.CNABFile, opts.Name, opts.Silent, opts.DryRun)
+	return err
+}
+
+func (p *Porter) chooseCredentialSet(CNABFile string) (string, error) {
+	var credSetNames []string
+	credSets, err := p.Credentials.ReadAll()
 	if err != nil {
-		return err
+		return "", errors.Wrap(err, "failed to read exisiting credential sets")
+	}
+
+	shouldGenerateCred := false
+	var selectedOption string
+	if len(credSets) > 0 {
+		for _, credSet := range credSets {
+			credSetNames = append(credSetNames, credSet.Name)
+		}
+
+		selectCredPrompt := &survey.Select{
+			Message: "Choose a set of credentials to use while installing this bundle",
+			Options: append(credSetNames, generateCredCode, quitGenerateCode),
+			Default: credSetNames[0],
+		}
+		survey.AskOne(selectCredPrompt, &selectedOption, nil)
+
+		switch selectedOption {
+		case generateCredCode:
+			shouldGenerateCred = true
+		case quitGenerateCode:
+			return "", errors.New("Credentials are mandatory to install this bundle")
+		default:
+			return selectedOption, nil
+		}
+	} else {
+		shouldGenerateCredPrompt := &survey.Confirm{
+			Message: "No credential identifier given. Generate one ?",
+		}
+		survey.AskOne(shouldGenerateCredPrompt, &shouldGenerateCred, nil)
+	}
+
+	if !shouldGenerateCred {
+		return "", errors.New("Credentials are mandatory to install this bundle")
 	}
-	name := opts.Name
-	if name == "" {
-		name = bundle.Name
+
+	return p.generateNewCredentialSet(CNABFile, "", false, false)
+}
+
+func (p *Porter) generateNewCredentialSet(CNABFile string, credIdentifierName string, silent bool, dryRun bool) (string, error) {
+	bundle, err := p.CNAB.LoadBundle(CNABFile)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to load bundle while generating credentials")
 	}
+
+	if credIdentifierName == "" {
+		if !silent {
+			inputCredNamePrompt := &survey.Input{
+				Message: "Enter credential identifier name",
+				Default: bundle.Name,
+			}
+			survey.AskOne(inputCredNamePrompt, &credIdentifierName, nil)
+		} else {
+			credIdentifierName = bundle.Name
+		}
+	}
+
 	genOpts := credentialsgenerator.GenerateOptions{
-		Name:        name,
+		Name:        credIdentifierName,
 		Credentials: bundle.Credentials,
-		Silent:      opts.Silent,
+		Silent:      silent,
+		DryRun:      dryRun,
+	}
+
+	fmt.Fprintf(p.Out, "  Generating new credential %s from bundle %s\n", genOpts.Name, bundle.Name)
+	fmt.Fprintf(p.Out, "  %d credentials required for bundle %s\n", len(genOpts.Credentials), bundle.Name)
+	err = p.generateAndSaveCredentialSet(genOpts)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to generate credentials")
 	}
-	fmt.Fprintf(p.Out, "Generating new credential %s from bundle %s\n", genOpts.Name, bundle.Name)
-	fmt.Fprintf(p.Out, "==> %d credentials required for bundle %s\n", len(genOpts.Credentials), bundle.Name)
+	fmt.Fprintln(p.Out, "Credentials generated and saved successfully for future use")
```

Comment:
We don't usually write such verbose success messages (or at all really). I'd either remove it or keep it short with

```suggestion
	fmt.Fprintf(p.Out, "Credentials saved as %s\n", credIdentifierName)
```